packages/reward-root-submitter/reward_root_submitter/main.py

```diff
@@ -1,58 +1,97 @@
-    "EVM_FULL_NODE_URL",
-    "OWNER",
-    "OWNER_PRIVATE_KEY",
-    "REWARD_POOL_ADDRESS",
-    "REWARD_PROGRAM_OUTPUT",
-    "ENVIRONMENT",
-]:
-    if expected_env not in os.environ:
-        raise ValueError(f"Missing environment variable {expected_env}")
-
-LOGLEVEL = os.environ.get("LOGLEVEL", "WARNING").upper()
-logging.basicConfig(level=LOGLEVEL)
-SENTRY_DSN = os.environ.get("SENTRY_DSN")
-if SENTRY_DSN is not None:
-    sentry_sdk.init(
-        SENTRY_DSN,
-        # Set traces_sample_rate to 1.0 to capture 100%
-        # of transactions for performance monitoring.
-        # We recommend adjusting this value in production.
-        traces_sample_rate=1.0,
-        environment=os.environ.get("ENVIRONMENT"),
+from .contracts import RewardPool
+
+NULL_HEX = HexBytes(
+    "0x0000000000000000000000000000000000000000000000000000000000000000"
+)
+
+EMPTY_MARKER_HEX = HexBytes(
+    "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+)
+
+
+@dataclass
+class MerkleRoot:
+    reward_program_id: str
+    payment_cycle: int
+    merkle_root_hash: HexBytes
+
+
+def setup_logging(config):
+    logging.basicConfig(level=config.log_level.upper())
+
+
+def safe_regex_group_search(regex, string, group):
+    """
+    Returns None in the case of a missing group
+    """
+    match = re.search(regex, string)
+    if match:
+        return match.group(group)
+    else:
+        return None
+
+
+def get_merkle_root_details(reward_output_filename):
+
+    reward_program_id = safe_regex_group_search(
+        r"rewardProgramID=([^/]*)", str(reward_output_filename), 1
     )
+    if not Web3().isChecksumAddress(reward_program_id):
+        raise Exception(
+            f"{reward_output_filename} does not have a valid checksummed address for the reward program ID"
+        )
+    payment_cycle = safe_regex_group_search(
+        r"paymentCycle=(\d*)", str(reward_output_filename), 1
+    )
+    if not (payment_cycle or "").isdigit():
+        raise Exception(
+            f"{reward_output_filename} does not have a valid payment cycle, should be not blank and a number"
+        )
+    with reward_output_filename.open("rb") as pf:
+        payment_file = pq.ParquetFile(pf)
+        # Read only a single row and a single column
+        try:
+            file_start = next(payment_file.iter_batches(batch_size=1))
+            first_row = file_start.to_pylist()[0]
+            root = HexBytes(first_row["root"])
+            if HexBytes(reward_program_id) != HexBytes(first_row["rewardProgramID"]):
+                raise Exception(
+                    f"{reward_output_filename} reward program ID in path and in the file do not match"
+                )
+            if payment_cycle != str(first_row["paymentCycle"]):
+                raise Exception(
+                    f"{reward_output_filename} payment cycle in path and in the file do not match"
+                )
+        except StopIteration:
+            root = EMPTY_MARKER_HEX
```

Comment:
Just wanna check theres no data issues here. 

- We do create EMPTY parquet files right? 
- All EMPTY parquet files are meant to have an EMPTY_MARKER_HEX written to the blockchain?

Reply:
> We do create EMPTY parquet files right?

Yes, currently when there are no payments, the file is empty. 

> All EMPTY parquet files are meant to have an EMPTY_MARKER_HEX written to the blockchain?

Yes.